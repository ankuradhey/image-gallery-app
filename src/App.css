header {
    background-color: #282c34;
    height: 48px;
    display: flex;
    flex-direction: row;
    align-items: center;
    color: white;
    padding: 20px;
    box-sizing: border-box;
}

.container {
    max-width: 1335px;
    margin: 0 auto;
}

.grid-row {
    display: flex;
    flex-flow: row wrap;
    justify-content: flex-start;
}

.grid-item {
    height: 350px;
    flex-basis: 20%;
    -ms-flex: auto;
    width: 259px;
    position: relative;
    padding: 10px;
    box-sizing: border-box;
}

.grid-item-wrapper {
    -webkit-box-sizing: initial;
    -moz-box-sizing: initial;
    box-sizing: initial;
    background: #ececec;
    margin: 0;
    height: 100%;
    width: 100%;
    overflow: hidden;
    -webkit-transition: padding 0.15s cubic-bezier(0.4, 0, 0.2, 1),
        margin 0.15s cubic-bezier(0.4, 0, 0.2, 1), box-shadow 0.15s cubic-bezier(0.4, 0, 0.2, 1);
    transition: padding 0.15s cubic-bezier(0.4, 0, 0.2, 1),
        margin 0.15s cubic-bezier(0.4, 0, 0.2, 1), box-shadow 0.15s cubic-bezier(0.4, 0, 0.2, 1);
    position: relative;
}

.grid-item-container {
    height: 100%;
    width: 100%;
    position: relative;
}
.grid-image-top {
    height: 80%;
    width: 120%;
    background-size: cover;
    position: relative;
    background-position: 50% 50%;
    left: -10.5%;
    top: -4.5%;
}

.wrapping-link {
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    z-index: 2;
}

.grid-item-content {
    padding: 0 20px 20px 20px;
}

.grid-item:hover .more-info i {
    padding-left: 20px;
    transition-duration: 0.5s;
}

.grid-item:hover .grid-item-wrapper {
    padding: 2% 2%;
    margin: -2% -2%;
}

@media (max-width: 1333px) {
    .grid-item {
        flex-basis: 33.33%;
    }
}

@media (max-width: 1073px) {
    .grid-item {
        flex-basis: 33.33%;
    }
}

@media (max-width: 815px) {
    .grid-item {
        flex-basis: 50%;
    }
}

@media (max-width: 555px) {
    .grid-item {
        flex-basis: 100%;
    }
}
