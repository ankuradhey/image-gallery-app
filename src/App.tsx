import React, { useState } from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { Gallery } from "./components/Gallery";
import { Header } from "./components/Header";
import { ImageDetail } from "./components";
import { AppContext } from "./context/AppContext";
import "./App.css";

function App() {
    const [gallery, setGallery] = useState([]);

    return (
        <div className="container">
            <Header />
            <AppContext.Provider
                value={{
                    gallery,
                    setGallery,
                }}
            >
                <Router>
                    <Switch>
                        <Route exact path="/" component={Gallery}></Route>
                        <Route exact path="/image/:id" component={ImageDetail}></Route>
                    </Switch>
                </Router>
            </AppContext.Provider>
        </div>
    );
}

export default App;
