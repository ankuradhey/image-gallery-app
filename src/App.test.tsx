import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";

describe("App Component", () => {
    beforeEach(() => {
        // IntersectionObserver isn't available in test environment
        const mockIntersectionObserver = jest.fn();
        mockIntersectionObserver.mockReturnValue({
            observe: () => null,
            unobserve: () => null,
            disconnect: () => null,
        });
        window.IntersectionObserver = mockIntersectionObserver;
    });

    it("Should render App component correctly", () => {
        const { container } = render(<App />);
        expect(container).toMatchSnapshot();
    });
});
