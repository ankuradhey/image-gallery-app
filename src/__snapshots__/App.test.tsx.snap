// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Should render App component correctly 1`] = `
<div>
  <div
    class="container"
  >
    <header>
      <a
        class="logo"
        href="/"
      >
        Image Gallery
      </a>
    </header>
    <section
      class="gallery"
    >
      <section
        class="filter"
      >
        <div
          class="filter__group"
        >
          <label
            class="filter__label"
          >
            Section:
          </label>
          <div
            class="filter__inputgroup"
          >
            <button
              class="filter__button--active"
            >
              HOT
            </button>
            <button
              class="filter__button"
            >
              TOP
            </button>
            <button
              class="filter__button"
            >
              USER
            </button>
          </div>
        </div>
        <div
          class="filter__group"
        >
          <label
            class="filter__label"
          >
            Window:
          </label>
          <div
            class="filter__inputgroup"
          >
            <button
              class="filter__button--active"
            >
              DAY
            </button>
            <button
              class="filter__button"
            >
              WEEK
            </button>
            <button
              class="filter__button"
            >
              MONTH
            </button>
            <button
              class="filter__button"
            >
              YEAR
            </button>
            <button
              class="filter__button"
            >
              ALL
            </button>
          </div>
        </div>
        <div
          class="filter__group"
        >
          <label
            class="filter__label"
          >
            Sort By:
          </label>
          <div
            class="filter__inputgroup"
          >
            <button
              class="filter__button--active"
            >
              VIRAL
            </button>
            <button
              class="filter__button"
            >
              TOP
            </button>
            <button
              class="filter__button"
            >
              TIME
            </button>
            <button
              class="filter__button"
            >
              RISING
            </button>
          </div>
        </div>
        <div
          class="fliter__group"
        >
          <label
            class="filter__label"
          >
            Show Viral Images:
          </label>
          <div
            class="filter__inputgroup"
          >
            <input
              type="checkbox"
              value="false"
            />
          </div>
        </div>
      </section>
      <main
        class="container"
      >
        <div
          class="content grid-row"
        />
      </main>
      <div
        class="loading loading--active"
        data-testid="loading"
      >
        <span>
          Loading...
        </span>
      </div>
    </section>
  </div>
</div>
`;
